packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1026,11 +1051,10 @@ class ThemeData extends Diagnosticable {
 
   @override
   int get hashCode {
-    // The hashValues() function supports up to 20 arguments.
-    return hashValues(
-      // Warning: make sure these properties are in the exact same order as in
-      // operator == and in the raw constructor and in the order of fields in
-      // the class and in the lerp() method.
+    // Warning: For the sanity of the reader, please make sure these properties
+    // are in the exact same order as in operator == and in the raw constructor
+    // and in the order of fields in the class and in the lerp() method.
+    final List<Object> values = <Object>[
```

Comment:
Why was the original code using the hashValues function instead of the hashList that you are using? Was it a performance concern or somesuch?

Reply:
No, it's not that,`hashList` just didn't exist when this `hashCode` function was first added.